onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -2779,6 +2779,24 @@ This op functions in much the same was as Dropout-11 and Dropout-13 do, execpt t
                       "Allow inputs and outputs to be any kind of tensor.");
 #endif
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(QuadricCustomOp)
+      .SetDomain(kQuadricDomain)
+      .SinceVersion(1)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+      .SetDoc("QuadricCustomOp")
+      .Input(0, "inputs", "QuadricCustomOp inputs.", "T", OpSchema::Variadic,
+             /*is_homogeneous*/ false,
+             /*min_arity*/ 1)
+      .Output(0, "outputs", "QuadricCustomOp outputs.", "T", OpSchema::Variadic,
+              /*is_homogeneous*/ false,
+              /*min_arity*/ 1)
+      .AllowUncheckedAttributes()
+      .Attr("ccl_func_name", "Name of CCL function.", AttributeProto::STRING)
+      .Attr("sub_graph", "Replaced sub-graph.", AttributeProto::GRAPH)
+      .TypeConstraint("T", OpSchema::all_tensor_types_with_bfloat(),
+                      "Allow inputs and outputs to be any kind of tensor.");
+  // FIXME: Add a type/shape inference function
```

Comment:
Do we need to preserve the `out_type` attribute until we add this? Or do we not care bc we assume shape inference has already been run prior to custom-op replacement?

Reply:
Yeah, we need to keep preserve it because we can't run shape inference in TVM on this op, and so therefore we need the `out_type` to tell us what the shape is.